src/offload_gemm.F

```diff
@@ -0,0 +1,395 @@
+!--------------------------------------------------------------------------------------------------!
+!   CP2K: A general program to perform molecular dynamics simulations                              !
+!   Copyright 2000-2022 CP2K developers group <https://cp2k.org>                                   !
+!                                                                                                  !
+!   SPDX-License-Identifier: GPL-2.0-or-later                                                      !
+!--------------------------------------------------------------------------------------------------!
+
+MODULE offload_gemm
```

Comment:
I'm a bit unhappy about the file name, because usually the prefix indicates the sub-directory. However, as you pointed out in https://github.com/cp2k/cp2k/pull/1951#issuecomment-1049811610, our `pdgemm` / `dgemm` code needs a bit of reorganization anyways. So, we can merge it like this and then I'll move things around in the coming weeks.